### ISSUE TITLE
Code running in WSL:Ubuntu will not load the codestyle.xml located in the Windows filesystem

### ISSUE THREAD

**Issue by rdcoe** [redhat-developer/vscode-java#1783]
I found that vscode cannot load an eclipse profile if vscode is using WSL as the terminal and the codestyle file is in the Windows filesystem.  This I found is true regardless of the url syntax for local files, i.e., D:\path\to\file or /d/path/to/file or file:///path/to/file, etc.  I had to copy the file into my WSL filesystem and reference is relative to the WSL root, i.e., /home/path/to/file.

##### Environment
- Operating System: Windows_NT x64 10.0.19042 + WSLv2 + Ubuntu 20.10
- JDK version: OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.20.10)
- Visual Studio Code version: 1.52.1, Electron 9.3.5 
- Java extension version:

##### Steps To Reproduce
1. configure Java › Format › Settings: Url to be drive_letter:\path\to\eclipse_codestyle.xml
2. confirm the settings aren't used when formatting file
3. configure Java › Format › Settings: Url to be file:///wsl_drive_mnt/path/to/eclipse_codestyle.xml
4. confirm the settings aren't used when formatting file
5. configure Java › Format › Settings: Url to be file:///path/to/eclipse_codestyle.xml
6. confirm file is used when formatting file

**Comment by snjeza** [comment]
@rdcoe you can try
/mnt/drive_letter/path/to/eclipse_codestyle.xml

**Comment by rdcoe** [comment]
I've changed the root mount point to use /, instead of /mnt/ so that's not
the problem.

**Comment by snjeza** [comment]
Could you try

ls /mnt/drive_letter/path/to/eclipse_codestyle.xml
or
ls /drive_letter/path/to/eclipse_codestyle.xml
for instance,
drive_letter is without : as /mnt/c/path/to/eclipse_codestyle.xml or /c/path/to/eclipse_codestyle.xml
See https://github.com/microsoft/WSL/issues/1918

**Comment by rdcoe** [comment]
> ls /d/OneDrive/Documents/Development/eclipse_codestyle.xml
/d/OneDrive/Documents/Development/eclipse_codestyle.xml*
> ls /mnt/d/OneDrive/Documents/Development/eclipse_codestyle.xml
ls: cannot access
'/mnt/d/OneDrive/Documents/Development/eclipse_codestyle.xml': No such file
or directory

Is this what you're looking for?

thx

**Comment by rdcoe** [comment]
btw, I see you linked it to the issue on changing the mount point.  I did that quite a long time ago, using the wsl.conf method that's described in the issue.

> cat /etc/wsl.conf
[automount]
root = /
options = "metadata"

**Comment by snjeza** [comment]
> Is this what you're looking for?

yes, it is. You should use:

```
"java.format.settings.url": "/d/OneDrive/Documents/Development/eclipse_codestyle.xml",
```
could you attach your [server log]( /d/OneDrive/Documents/Development/eclipse_codestyle.xml) ?

**Comment by rdcoe** [comment]
K, I just re-tested, so as to get new log data, and it worked.  Not sure
what was different between now and before, when I had previously defined
the url using a windows path, from when I ran code under a windows
profile.  When I changed it to use a WSL path, it didn't work.  Perhaps I
had a syntax error in my attempts to get it to work; I wasn't sure if I
needed a raw path or a file:/// URL reference, so maybe I messed that up?

Long story short, looks like this can be closed.  But perhaps a more
instructional message on this page,
https://github.com/redhat-developer/vscode-java/wiki/Formatter-settings,
which only has a remote http reference as an example would be helpful?

Thanks.
